liquidhaskell-boot/src/Language/Haskell/Liquid/GHC/Plugin.hs

```diff
@@ -256,12 +264,15 @@ typecheckHook cfg0 modSummary0 tcGblEnv = do
     thisModule :: Module
     thisModule = tcg_mod tcGblEnv
 
-    dropPlugins hsc_env = hsc_env { hsc_plugins = emptyPlugins }
-
     continue = pure $ Left (ErrorsOccurred [])
 
     updateModSummaryDynFlags f ms = ms { ms_hspp_opts = f (ms_hspp_opts ms) }
 
+isRecursiveChecking :: IORef Bool
+isRecursiveChecking = unsafePerformIO $ newIORef False
```

Comment:
Thanks again @gergoerdi.

Similar to #2367, to account for concurrency this should use a Set, where modules are added before they are typechecked by LH, and then they are removed after they have been typechecked.

I'd rename `isRecursiveChecking` to `isLHTypechecking`, and maybe place the explanation of the need to break an infinite loop at this point.